packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/migration/MigrationParticipantTests.java

```diff
@@ -0,0 +1,200 @@
+                    assertThat(migrationData.getLastMigrationPerformed()).isEqualTo(UpdateDiagramDescriptionNameValue.class.getSimpleName());
+                });
+
+            });
+        }
+    }
+
+    private static class UpdateLabelExpressionMigrationParticipant implements IMigrationParticipant {
+
+        private static final String PARTICIPANT_VERSION = "2024.4.0";
+
+        @Override
+        public String getVersion() {
+            return PARTICIPANT_VERSION;
+        }
+
+        @Override
+        public void getJSonObjectBeforeLoading(EObject eobject, JsonObject object) {
+            if (eobject instanceof NodeDescription nodeDescription) {
+               var nodeDescriptionData = object.getAsJsonObject("data");
+               var labelExpression = nodeDescriptionData.get("labelExpression").getAsString();
+               if (nodeDescription.getStyle() instanceof RectangularNodeStyleDescription) {
+                   InsideLabelDescription insideLabelDescription = DiagramPackage.eINSTANCE.getDiagramFactory().createInsideLabelDescription();
+                   insideLabelDescription.setLabelExpression(labelExpression);
+                   nodeDescription.setInsideLabel(insideLabelDescription);
+               }
+                if (nodeDescription.getStyle() instanceof ImageNodeStyleDescription) {
+                    OutsideLabelDescription outsideLabelDescription = DiagramPackage.eINSTANCE.getDiagramFactory().createOutsideLabelDescription();
+                    outsideLabelDescription.setLabelExpression(labelExpression);
+                    nodeDescription.getOutsideLabels().add(outsideLabelDescription);
+                }
+            }
+        }
+
+        @Override
+        public EStructuralFeature getElement(EClass eClass, String oldFeatureName) {
+            if (eClass == DiagramPackage.eINSTANCE.getDiagramDescription()) {
+                if (oldFeatureName.equals("nodeDescriptionsWrong")) {
+                    return DiagramPackage.eINSTANCE.getDiagramDescription_NodeDescriptions();
+                }
+            }
+
+            return IMigrationParticipant.super.getElement(eClass, oldFeatureName);
+        }
+    }
+
+    private static class UpdateNodeDescriptionDomainTypeFeatureName implements IMigrationParticipant {
+        private static final String PARTICIPANT_VERSION = "2024.3.0";
+
+        @Override
+        public String getVersion() {
+            return PARTICIPANT_VERSION;
+        }
+
+        @Override
+        public EStructuralFeature getElement(EClass eClass, String oldFeatureName) {
+            if (eClass == DiagramPackage.eINSTANCE.getNodeDescription()) {
+                if (oldFeatureName.equals("domainTypeWrong")) {
+                    return DiagramPackage.eINSTANCE.getDiagramElementDescription_DomainType();
+                }
+            }
+            return IMigrationParticipant.super.getElement(eClass, oldFeatureName);
+        }
+    }
```

Comment:
No need to create fake migration participants, I have plenty of real use cases that I want to support